File: module_gan/src/main/java/com/huyingbao/module/gan/ui/main/store/MainStore.java
package com.huyingbao.module.gan.ui.main.store;

import com.huyingbao.core.arch.dispatcher.RxDispatcher;
import com.huyingbao.core.arch.model.RxAction;
import com.huyingbao.core.arch.model.RxChange;
import com.huyingbao.core.arch.store.RxActivityStore;
import com.huyingbao.module.gan.ui.main.action.MainAction;

import org.greenrobot.eventbus.Subscribe;

import javax.inject.Inject;
import javax.inject.Singleton;

/**
 * Created by liujunfeng on 2019/1/1.
 */
@Singleton
public class MainStore extends RxActivityStore {
    @Inject
    MainStore(RxDispatcher rxDispatcher) {
        super(rxDispatcher);
    }

    /**
     * 接收RxAction
     * 处理RxAction携带的数据
     * 发送RxChange通知RxView
     *
     * @param rxAction
     */
    @Override
    @Subscribe()
    public void onRxAction(RxAction rxAction) {
        switch (rxAction.getTag()) {
            case MainAction.TO_WAN_MODULE:
            case MainAction.TO_GAN_MODULE:
                postChange(RxChange.newInstance(rxAction));
                break;
        }
    }
}

File: module-github/src/main/java/com/huyingbao/module/github/ui/main/store/MainStore.kt
package com.huyingbao.module.github.ui.main.store

import androidx.lifecycle.MutableLiveData
import com.huyingbao.core.arch.dispatcher.RxDispatcher
import com.huyingbao.core.arch.model.RxAction
import com.huyingbao.core.arch.store.RxActivityStore
import com.huyingbao.module.github.database.GithubAppDatabase
import com.huyingbao.module.github.ui.main.action.MainAction
import com.huyingbao.module.github.ui.main.model.Event
import com.huyingbao.module.github.ui.main.model.Repos
import org.greenrobot.eventbus.Subscribe
import org.greenrobot.eventbus.ThreadMode
import javax.inject.Inject
import javax.inject.Singleton

/**
 * 主页模块
 *
 * Created by liujunfeng on 2019/6/10.
 */
@Singleton
class MainStore @Inject constructor(
        rxDispatcher: RxDispatcher,
        private var githubAppDatabase: GithubAppDatabase
) : RxActivityStore(rxDispatcher) {
    /**
     * 动态事件数据
     */
    val eventListLiveData = MutableLiveData<ArrayList<Event>>()
    /**
     * 推荐趋势仓库数据
     */
    val trendListLiveData = githubAppDatabase.plantDao().getReposListLiveData()

    override fun onCleared() {
        super.onCleared()
        eventListLiveData.value = null
    }

    @Subscribe(tags = [MainAction.GET_NEWS_EVENT])
    fun onGetNewsEvent(rxAction: RxAction) {
        eventListLiveData.value = rxAction.getResponse()
    }

    @Subscribe(tags = [MainAction.GET_TREND_DATA], threadMode = ThreadMode.POSTING)
    fun onGetTrend(rxAction: RxAction) {
        val plants = rxAction.getResponse<List<Repos>>()
        if (plants != null) {
            githubAppDatabase.plantDao().insertAll(plants)
        }
    }
}